integrations/gitlab_v2/rate_limiter.py

```diff
@@ -0,0 +1,45 @@
+import asyncio
+import time
+from loguru import logger
+
+class GitLabRateLimiter:
```

Comment:
Can you look into handling the acquire lock and release with a while loop over a Bounded Semaphore to abstract this and not reinvent the wheel. Can you explain why you chose this approach?